pipeline/util/goa_filter

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/awk -f
+# WB	WBGene00000018	abl-1		GO:1901075	WB_REF:WBPaper00033098|PMID:19402756	IGI	WB:WBGene00000421	P		M79.1	gene	taxon:6239	20120809	WB
+BEGIN {
+    FS="\t"
+}
+{
+    if (index($0, "!") == 1) {
+        # Pass through comments
+        print $0
+        next
+    }
+
+    if (NF != 15 && NF != 17) {
+        # Skip if not the correct number of columns
+        # print "skipping, cause bad columns " NF
+        next
+    }
+
+    if ($7 == "IEA") {
+        # Skip line if IEA
+        # print "skipping iea"
+        next
+    }
+
+    taxa = $13
+    if(index($13, "|") > 0) {
+        taxa = substr(taxa, 1, index($13, "|")-1)
+    }
+
+    if (taxa in model_org_cache) {
+        # taxon is a model organism, so we skip it
+        # print "cache: is model organism"
+        next
+    } else if (taxa in goa_cache) {
+        # taxon is not a model organism, so we print it
+        # print "cache: is not a model organism"
+        print $0
+        next
+    } else {
+        # print "taxon not in caches, finding it json"
+        command="python util/model_organism.py --silent " taxa " ../metadata/datasets"
```

Comment:
python, not python3 or path? I can see this might be difficult in a jenkins environment...

Reply:
Would that interpreter run every time?

Reply:
I had envisioned a one time compilation step to generate the awk or something similar

Reply:
No this isn't run every time. It saves off in an array taxons it has seen already, and checks there first. Really though, I think this could be simplified with little different approach. This was just my first pass. I was thinking about changing this so that there was a make target to find all the model organism taxa, and then we could load that file here one time at the start, and check against that. I think that is ultimately a cleaner and faster solution, I just hadn't gotten to that yet. That would also take care of the interpreter path issue. And you're right, jenkins could have a difficult time with that.